cpp/src/arrow/util/bit_stream_utils.h

```diff
@@ -324,11 +324,12 @@ inline int BitReader::GetBatch(int num_bits, T* v, int batch_size) {
   int max_bytes = max_bytes_;
   const uint8_t* buffer = buffer_;
 
-  uint64_t needed_bits = num_bits * batch_size;
+  const int64_t needed_bits = num_bits * static_cast<int64_t>(batch_size);
```

Comment:
not clear from the context but do we need to worry about overflow as well?

Reply:
`batch_size` is originally an int and I don't think `num_bits` is ever greater than 64, though I don't know for sure.